runtime/backend/interface.cpp

```diff
@@ -9,21 +9,86 @@
 #include <executorch/runtime/backend/interface.h>
 #include <executorch/runtime/platform/assert.h>
 
-namespace torch {
-namespace executor {
-
-PyTorchBackendInterface::~PyTorchBackendInterface() {}
+#ifdef _WIN32
+#include <memory>
+#include <windows.h>
+#include <tchar.h>
+#define getpid GetCurrentProcessId
+#else
+#include <unistd.h>
+#endif
 
 // Task t128866626: Remove global static variables.
 // We want to be able to run multiple Executor instances
 // and having a global registration isn't a viable solution
 // in the long term.
-BackendRegistry& getBackendRegistry();
-BackendRegistry& getBackendRegistry() {
-  static BackendRegistry backend_reg;
+#ifdef _WIN32
+
+#define SHARED_MEMORY_NAME "torch_executor_backend_registry"
+static std::shared_ptr<torch::executor::BackendRegistry> backend_reg;
+
+torch::executor::BackendRegistry& getBackendRegistry() {
```

Comment:
Because Windows require mmap to share memory between DLLs.
https://learn.microsoft.com/en-us/windows/win32/dlls/using-shared-memory-in-a-dynamic-link-library

Without these change.
For example 
`llama_main.exe` load `libextension_module.dll` and `libexecutorch.dll`
`libextension_module.dll` load `libexecutorch.dll`
Will result the registry which `llama_main.exe` uses have all registered element but `libextension_module.dll` not

Reply:
I found shared section could help, not need mmap